metrics/group_coordinators.go

```diff
@@ -29,11 +29,11 @@ func NewCoordinatorMetricGroup(info *MetricInfo) *CoordinatorMetricGroup {
 
 func (c *CoordinatorMetricGroup) GetMetrics(status *models.FullStatus) {
 	scope := c.GetScopeOrExit("default")
-	if status == nil || status.Client == nil || status.Client.Coordinators == nil {
+	if !isValidClient(status) || status.Client.Coordinators == nil {
```

Comment:
Would you consider implementing a Validate() method on `models.FullStatus` so that if it gets extended in the future its validation will be right there as well?

Reply:
I tried this, and the code became much more complex. I think it's better to keep the model as the model and not split the metrics generation logic with the model and the metrics packages.